add guidance to howto form

benfurber opened the pull request:
PR Checklist

- [x] - Commit [messages are descriptive](https://github.com/ONEARMY/community-platform/blob/master/CONTRIBUTING.md#--commit-style-guide), it will be used in our [Release Notes](https://github.com/ONEARMY/community-platform/releases/)

PR Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Developer experience (improves developer workflows for contributing to the project)

## Description

This adds guidance for the user on certain how-to categories in which additional context is provided.

## Git Issues

Closes #2683 

## Screenshots/Videos

<img width="731" alt="Screenshot 2023-09-26 at 16 17 33" src="https://github.com/ONEARMY/community-platform/assets/16688508/db72f6f7-5c82-4c03-968b-8ecac60ef97c">
<img width="731" alt="Screenshot 2023-09-26 at 16 17 50" src="https://github.com/ONEARMY/community-platform/assets/16688508/976f79ac-7c89-47d9-b697-16cced1f2c71">

thisislawatts commented on src/pages/Howto/Content/Common/HowtoCategoryGuidance.test.tsx:
```diff
@@ -0,0 +1,49 @@
+      </HowtoProvider>,
+    )
+
+    await screen.findByText('moulds category have some requirements', {
```
**question:** Could we load the `findByText` from the labels file? This would remove the hard coded text and make the tests less brittle to copy changes.

benfurber replied:
I like the approach. Seeing as this label is formatted into html it makes this specific test still brittle but maybe less brittle?

benfurber commented:
@thisislawatts Suggested test change made with a comment. Also realising because of the order of work I did this month that this branch is pulling in code that's for #2720 - apologies.

thisislawatts commented:
> @thisislawatts Suggested test change made with a comment. Also realising because of the order of work I did this month that this branch is pulling in code that's for #2720 - apologies.

Gotcha, I think let's focus on merging #2720 as that is a subset of this PR and you and @evakill are working through review atm. I will label this one as blocked for the time being and we can come back to it after #2720 has been merged.

thisislawatts commented:
@benfurber, now that #2720 is merged do you think you can rebase this PR and we can move it forwards?

benfurber commented:
@thisislawatts Absolutely. Ready to finish this off. Added a clarifying question on #2683.

thisislawatts commented on src/pages/Howto/labels.ts:
```diff
@@ -85,6 +85,58 @@ export const intro: ILabels = {
+      '<ul><li>Step 1 > Explain the Guide</li>' +
+      '<li>Step 2 (as many steps as needed) > Talk about the process</li>' +
+      '<li>Step 3 (as many steps as needed) > Talk about the challenges</li>' +
+      '<li>Step 5 > Tips & tricks</li>' +
```
**question:** Why have we marked this up as an unordered list with `ul` but then added a `Step n` to each one. Would an ordered list `ol` better align with the semantics of the content?

thisislawatts replied:
**issue(blocking):** We are missing a Step 4 from this list.

thisislawatts commented on src/pages/Howto/labels.ts:
```diff
@@ -85,6 +85,58 @@ export const intro: ILabels = {
+      '<ul><li>Step 1 > Explain the product</li>' +
+      '<li>Step 2 (as many steps as needed) > Talk about the process</li>' +
+      '<li>Step 3 (as many steps as needed) > Challenges making the product</li>' +
+      '<li>Step 5 > Tips & tricks</li>' +
```
issue(blocking): We are missing a Step 4 from this list.

benfurber commented:
@thisismattia and @sigolenej - As I copied the copy exactly from [the guide](https://community.preciousplastic.com/academy/create/howto), @thisislawatts has a couple of questions for you.

1. As the steps are numbered and start 'step 1', etc. Would it be better to use a list with numbers rather than bullet points.
2. Products and guides 'miss' step 4 which I assumed is because you're trying to stay 'from step 5 onwards', but maybe that is a typo?

thisismattia commented:
Hey @benfurber nice one!

1. Yep ordered list sounds good
2. Defo a typo :)

benfurber commented:
Fixed. Updated screenshot:
<img width="550" alt="Screenshot 2023-09-27 at 16 12 43" src="https://github.com/ONEARMY/community-platform/assets/16688508/dac3ce91-a501-4692-b4a8-0aab535af5e7">

thisislawatts commented:
Thanks @benfurber, looks great. Lets get this merged. I am curious does this also resolve Issue #2684?
